### ISSUE TITLE
Disassembly view: do not use piggy back on existing step_over commands, introduce new ones

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#129530]
fyi @yuehuang010, @xisui-MSFT 

Code pointer https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/debugCommands.ts#L265

I already mentioned this in the PR review but it seems like we did not address this.

We should not check the value of the context inside the command, but we should have special STEP commands for the disassembly view.
That way the solution will be cleaner.

**Comment by xisui-MSFT** [comment]
Thanks for the feedback. The reason I changed this back this way is step commands are registered in the debug tool widget in a way that seems can only accept one command for each button. And I was not sure how to make the stepping command in debug tool decide to do instruction stepping or source line stepping. Any suggestions?

**Comment by isidorn** [comment]
Yeah I think that the current step commands need to change their context key such that they are **not** registered when the disassembly view is focused.
And when it is focused that your new commands get registered instead.

I think we should not tackle this this week, so assigning to next milestone when you can hopefully look into this.
